flink-ml-lib/src/main/java/org/apache/flink/ml/recommendation/swing/Swing.java

```diff
@@ -0,0 +1,448 @@
+        final LogicalType userColType =
+                schema.getColumn(userCol).get().getDataType().getLogicalType();
+        final LogicalType itemColType =
+                schema.getColumn(itemCol).get().getDataType().getLogicalType();
+
+        if ((!TypeCheckUtils.isLong(InternalTypeInfo.of(userColType).toLogicalType())
+                        && !TypeCheckUtils.isInteger(
+                                InternalTypeInfo.of(userColType).toLogicalType()))
+                || (!TypeCheckUtils.isLong(InternalTypeInfo.of(itemColType).toLogicalType())
+                        && !TypeCheckUtils.isInteger(
+                                InternalTypeInfo.of(itemColType).toLogicalType()))) {
+            throw new IllegalArgumentException("Type of user and item column must be long or int.");
+        }
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        SingleOutputStreamOperator<Tuple2<Long, Long>> itemUsers =
+                tEnv.toDataStream(inputs[0])
+                        .map(
+                                row ->
+                                        Tuple2.of(
+                                                ((Number) row.getFieldAs(userCol)).longValue(),
+                                                ((Number) row.getFieldAs(itemCol)).longValue()))
+                        .returns(Types.TUPLE(Types.LONG, Types.LONG));
+
+        SingleOutputStreamOperator<Tuple3<Long, Long, List<Long>>> userAllItemsStream =
+                itemUsers
+                        .keyBy(tuple -> tuple.f0)
+                        .transform(
+                                "fillUserItemsTable",
+                                Types.TUPLE(Types.LONG, Types.LONG, Types.LIST(Types.LONG)),
+                                new BuildSwingData(getMinUserItems(), getMaxUserItems()));
+
+        SingleOutputStreamOperator<Tuple3<Long, Long, Double>> similarity =
+                userAllItemsStream
+                        .keyBy(tuple -> tuple.f1)
+                        .transform(
+                                "calculateSwingSimilarity",
+                                Types.TUPLE(Types.LONG, Types.LONG, Types.DOUBLE),
+                                new CalculateSimilarity(
+                                        getK(),
+                                        getMaxItemUsers(),
+                                        getAlpha1(),
+                                        getAlpha2(),
+                                        getBeta()));
+
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        new TypeInformation[] {
+                            BasicTypeInfo.LONG_TYPE_INFO,
+                            Types.TUPLE(
+                                    BasicTypeInfo.LONG_TYPE_INFO, BasicTypeInfo.DOUBLE_TYPE_INFO)
+                        },
```

Comment:
I think one output row need contains the item and the recommended info which includes recommended items and the scores.